pkg/network/http/monitor.go

```diff
@@ -90,70 +82,50 @@ func (http *Monitor) Start() error {
 	}
 
 	go func() {
-		var (
-			misses int
-			then   = time.Now()
-			hits   = make(map[int]int)
-			report = time.NewTicker(30 * time.Second)
-			key    = new(httpBatchKey)
-		)
+		report := time.NewTicker(30 * time.Second)
 		defer report.Stop()
-
 		for {
 			select {
 			case data, ok := <-http.perfHandler.DataChannel:
 				if !ok {
 					return
 				}
 
-				// The notification we read off the perf ring tells us which HTTP batch of transactions is ready to be read
+				// The notification we read from the perf ring tells us which HTTP batch of transactions is ready to be consumed
 				notification := toHTTPNotification(data)
-				key.Prepare(notification)
-				batch := new(httpBatch)
-				err := http.batchMap.Lookup(unsafe.Pointer(key), unsafe.Pointer(batch))
-				if err != nil {
-					log.Errorf("error retrieving http batch for cpu=%d", notification.cpu)
-					continue
-				}
-
-				if batch.IsDirty(notification) {
-					misses++
-					continue
-				}
-
-				txs := batch.GetTransactions(notification)
-				// This is where we would add code handling the HTTP data (eg., generating latency percentiles etc.)
-				// Right now I'm just aggregating the hits per status code just as a placeholder to make sure everything
-				// is working as expected
-				for _, tx := range txs {
-					hits[tx.StatusClass()]++
-				}
+				transactions, err := http.batchManager.GetTransactionsFrom(notification)
+				http.process(transactions, err)
 			case _, ok := <-http.perfHandler.LostChannel:
 				if !ok {
 					return
 				}
-				misses++
-			case now := <-report.C:
-				delta := float64(now.Sub(then).Seconds())
-				log.Infof("http report: 100(%d reqs, %.2f/s) 200(%d reqs, %.2f/s) 300(%d reqs, %.2f/s), 400(%d reqs, %.2f/s) 500(%d reqs, %.2f/s), misses(%d reqs, %.2f/s)",
-					hits[100], float64(hits[100])/delta,
-					hits[200], float64(hits[200])/delta,
-					hits[300], float64(hits[300])/delta,
-					hits[400], float64(hits[400])/delta,
-					hits[500], float64(hits[500])/delta,
-					misses*HTTPBatchSize,
-					float64(misses*HTTPBatchSize)/delta,
-				)
-				hits = make(map[int]int)
-				then = now
-				misses = 0
+
+				http.process(nil, errLostBatch)
+			case reply := <-http.pollRequests:
+				transactions := http.batchManager.GetPendingTransactions()
+				http.process(transactions, nil)
+				reply <- struct{}{}
+			case <-report.C:
+				transactions := http.batchManager.GetPendingTransactions()
+				http.process(transactions, nil)
+				http.telemetry.report()
 			}
 		}
 	}()
 
 	return nil
 }
 
+// Sync HTTP data between userspace and kernel space
+func (http *Monitor) Sync() {
+	reply := make(chan struct{}, 1)
+	defer close(reply)
+
+	// TODO: Add logic to ensure this won't deadlock during termination
```

Comment:
Maybe this should be resolved before merging?

Reply:
Agreed! Done